internal/services/communication/email_service_domain_sender_username_resource.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package communication
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/communication/2023-03-31/senderusernames"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.ResourceWithUpdate = EmailCommunicationServiceDomainSenderUsernameResource{}
+
+type EmailCommunicationServiceDomainSenderUsernameResource struct{}
+
+type EmailCommunicationServiceDomainSenderUsernameModel struct {
+	Name                 string `tfschema:"name"`
+	EMailServiceDomainID string `tfschema:"email_service_domain_id"`
+	DisplayName          string `tfschema:"display_name"`
+	Username             string `tfschema:"username"`
+}
+
+func (EmailCommunicationServiceDomainSenderUsernameResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringLenBetween(1, 253),
+		},
+
+		"email_service_domain_id": {
```

Comment:
consider using `consider using the commonschema.ResourceIDReferenceForceNew` for `email_service_domain_id`

Reply:
Updated